src/main.rs

```diff
@@ -188,20 +235,18 @@ fn wait_fs(rx: &Receiver<Event>, filter: &NotificationFilter) -> Vec<PathBuf> {
     paths
 }
 
-fn wait_process(process: &RwLock<Option<Process>>, kill: bool, restart: bool) {
+// wait_process sends signal to process. It waits for the process to exit if wait is true
+fn wait_process(process: &RwLock<Option<Process>>, signal: Option<Signal>, wait: bool) {
```

Comment:
If this doesn't always wait we should give this a clearer name, such as `signal_process()`.